File: filterls/filter.py
#!/usr/bin/env python3

import sys

def filter(line):
    segs = line.split(maxsplit=2)
    if len(segs) > 1 and segs[1].isnumeric():
        return int(segs[1]) > 10

def process():
    for line in sys.stdin:
        if filter(line):
            print(line[:-1])

if __name__ == "__main__":
    process()

File: filterls/filter.js
#!/usr/bin/env node

const readline = require("readline")

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

const filterLine = line => {
    const parts = line.split(" ", 3);
    if (parts.length < 2) { return false; }
    const n = parseInt(parts[1]);
    if (isNaN(n) || n <= 10) { return false; }
    return true;
};

rl.on("line", line => {
    if (filterLine(line)) { console.log(line); }
});

